src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1633,12 +1633,16 @@ private NonReentrantLock StateLock
         ///
         /// This not intended to be the public API, use Document.WithFrozenPartialSemantics() instead.
         /// </summary>
-        public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, CancellationToken cancellationToken)
+        public async ValueTask<SolutionState> WithFrozenPartialCompilationIncludingSpecificDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
         {
             try
             {
+                // Ensure we explicitly move to the BG so that if we do any IO it does not potentially happen
+                // on a UI thread caller.
+                await Task.Yield().ConfigureAwait(false);
+
                 var doc = this.GetRequiredDocumentState(documentId);
-                var tree = doc.GetSyntaxTree(cancellationToken);
+                var tree = await doc.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
 
                 using (this.StateLock.DisposableWait(cancellationToken))
```

Comment:
📝 This should use the async form in asynchronous code

Reply:
This is a NonReentrantLock, which doesn't currently have the async codepath :)